scRNA-seq/02-normalizing_scRNA-seq.Rmd

```diff
@@ -67,185 +72,355 @@ sc_metadata <- readr::read_tsv(metadata_file)
+This is not the default for `calculatePCA()` as we will explore below.
 
-# Set up the normalized PC scores in the same way
-norm_pca_scores <- data.frame(norm_pca$x[, 1:2],
-                              geo_accession = rownames(norm_pca$x)) %>%
-  dplyr::left_join(dplyr::select(sc_metadata, 
-                                 geo_accession, 
-                                 cell_type = cell.type.ch1))
+Next we will arrange the PCA scores for plotting, adding a column with the cell-type data so we can color each point of the plot. 
+
+```{r pca_df}
+# Set up the PCA scores for plotting
+norm_pca_scores <- data.frame(norm_pca,
+                              geo_accession = rownames(norm_pca),
+                              cell_type = gbm_sce$cell_type)
+log_pca_scores <- data.frame(log_pca,
+                             geo_accession = rownames(log_pca),
+                             cell_type = gbm_sce$cell_type)
 ```
 
-Let's plot the counts PCA scores with their cell labels:
+Now we will plot the unnormalized PCA scores with their cell labels:
 
-```{r Plotting counts pca}
+```{r pca_plot}
 # Now plot counts pca
-ggplot(count_pca_scores, aes(x = PC1, y = PC2, color = cell_type)) +
+ggplot(log_pca_scores, aes(x = PC1, y = PC2, color = cell_type)) +
   geom_point() + 
-  ggtitle("Counts PCA Scores") + 
+  ggtitle("Log counts (unnormalized) PCA scores") + 
   colorblindr::scale_color_OkabeIto() # We are adding this so it is colorblind friendly
 ```
 
-We've plotted the counts data for you. Knowing that we want the same graph,
-but different data, use the above template to plot the normalized data.
+We've plotted the counts data for you. Knowing that we want the same graph, but different data, use the above template to plot the normalized data. 
+Feel free to customize the plot with a different theme or color scheme!
 
 Let's plot the `norm_pca_scores` data:
 
-```{r Plotting norms pca, live = TRUE}
+```{r norm_pca_plot, live = TRUE}
 ggplot(norm_pca_scores, aes(x = PC1, y = PC2, color = cell_type)) +
   geom_point() + 
-  ggtitle("Normalized log counts PCA Scores") + 
+  ggtitle("Normalized log counts PCA scores") + 
+  colorblindr::scale_color_OkabeIto()
+```
+
+Do you see an effect from the normalization in the comparison between these plots?
+
+## Modeling variance
+
+The variation in gene expression we see among cells comes from a combination of variation due to technical effects and the biology we really care about. 
+In order to account for this we could just take the largest variance genes, but if we want to be a bit more careful about it, we can model the variance in expression of each gene as a function of mean expression for that gene.
+This is useful because we generally expect the variance to increase as mean expression increases, even if there is no biological signal in the expression variation.
+
+We will do this modeling of variance by expressionwith the `scran::modelGeneVar()` function, saving the results to a new variable. 
```

Comment:
```suggestion
We will do this modeling of variance by expression with the `scran::modelGeneVar()` function, saving the results to a new variable. 
```

Quoting from the docs for `modelGeneVar()` for note to self reasons

> For each gene, we compute the variance and mean of the log-expression values. A trend is fitted to the variance against the mean for all genes using fitTrendVar. The fitted value for each gene is used as a proxy for the technical component of variation for each gene, under the assumption that most genes exhibit a low baseline level of variation that is not biologically interesting. The biological component of variation for each gene is defined as the the residual from the trend.

Reply:
Yeah, I will admit I don't love this model as a representation of the two components of variance. There is another modelGeneVarPoisson() (or something like), which seems like it might be more reasonable. But I like the point that doing this modeling makes: not all the variation we see is biological, and there is a relationship between mean expression and variation that we should keep ourselves aware of.